File: find_the_pivot_integer.go
func pivotInteger(n int) int {
    sum, curr := n*(n+1)/2, 0;

    for i:=1; i <= n; i++ {
        curr += i;
        if(curr == sum-curr+i) {
            return i;
        }
    }
    return -1;
}

// Time Complexity : O(n)
// Space Complexity : O(1)

File: find_the_pivot_integer.cpp
class Solution {
public:
    int pivotInteger(int n) {
        int sum = n*(n+1)/2;
        int curr = 0;

        for(int i=1; i <= n; i++) {
            curr += i;
            if(curr == sum-curr+i) return i;
        }
        return -1;
    }
};

// Time Complexity : O(n)
// Space Complexity : O(1)